### ISSUE TITLE
Toggling EW ROI gives traceback

### ISSUE THREAD

**Issue by pllim** [spacetelescope/specviz#167]
Again, this is with Gisella's MDM data. All I did was to toggle the ROI selection for EW measurement.

```python
ERROR:root:Traceback (most recent call last):
  File ".../specviz/core/comms.py", line 112, in wrapper
    return func(*args, **kwargs)
  File ".../specviz/ui/controller.py", line 556, in update_statistics
    cent = statistics.centroid(line - avg_cont)
  File ".../specviz/core/data.py", line 202, in __sub__
    new_source = self._arithmetic(self._source.subtract, other)
  File ".../specviz/core/data.py", line 164, in _arithmetic
    other = self._source._from_self(other)
  File ".../specviz/core/data.py", line 76, in _from_self
    dispersion_unit=self.dispersion_unit)
  File ".../specviz/core/data.py", line 52, in __init__
    super(Data, self).__init__(data=data, *args, **kwargs)
  File ".../astropy/nddata/nddata.py", line 159, in __init__
    raise ValueError("Cannot use the unit argument when data "
ValueError: Cannot use the unit argument when data is a Quantity
```

**Comment by nmearl** [comment]
Which MDM data? I can't reproduce on the first couple of ones in the directory.

**Comment by pllim** [comment]
The first one, `120201`. Does not matter whether I use Astropy 1.1.1 or the latest dev.

**Comment by pllim** [comment]
More info: I can see the ROI being shown, but no EW measurements are computed successfully; i.e., the labels for results are all blank.

**Comment by nmearl** [comment]
I still cannot reproduce, but I think I might have been able to avoid the error from your traceback. Can you pull from master and give it a shot?

**Comment by pllim** [comment]
It's so weird that I get all these tracebacks that no one else is getting! Thanks, I'll follow up next week.

**Comment by pllim** [comment]
http://docs.astropy.org/en/stable/analytic_functions/index.html#using-astropy-analytic-functions

I hacked together a `Blackbody` model for `pysynphot` a while back but that project is on hold.

**Comment by pllim** [comment]
@nmearl , your fix didn't work for me, but it gave me a hint where to look. I fixed it in #172. The only explanation that makes sense is somehow my setup causes one of the functions to return scalar `Quantity`.

**Comment by nmearl** [comment]
:+1:
